setup.cfg

```diff
@@ -30,6 +30,7 @@ packages = find:
 zip_safe = false
 install_requires =
     click >= 7
+    pep517
```

Comment:
Done!

Reply:
@atugushev using private things is usually not the best idea. By depending on `pep517` we could pull-in more improvements earlier than pip and this would work better when some old pip version is used.

Reply:
> this would work better when some old pip version is used.

This should not be an issue since we support only the latest two minor versions of `pip` by convention.

Reply:
Still, vendored libs aren't public API. They can update them even in patch releases. This alone is a reason not to rely on private parts.

Reply:
Other than the CLI, none of pip is public. The pip maintainers are very clear about that.

Since we're already ignoring that, I agree with @atugushev that we might as well leverage the vendored libraries too. It reduces the number of hard dependencies necessary to use pip-tools

Reply:
Friendly ping :) Is there consensus on this?

Reply:
I agree with @jdufresne. Yes, it is a bad practice to use vendored dependencies, especially from another library, especially underscored. However, pip-tools is already dangerously dependent on pip intrinsics, so not a big deal. Also, it's unlikely that `pep517` will drastically change its API or pip will unvendorize it.

Reply:
FYI https://github.com/jazzband/pip-tools/issues/1352

Reply:
FTR I just remembered another case: Ubuntu/Debian decouple system pip's vendored libs into separate packages so their importable paths change.

Reply:
The use of `pip._vendor.pep517` is being reverted in https://github.com/jazzband/pip-tools/pull/1353 in favor of declaring the `pep517` dependency. Resolving this conversation.